src/client.cpp

```diff
@@ -291,31 +281,89 @@ void CClient::OnNewConnection()
     //### TODO: END ###//
 }
 
-void CClient::OnConClientListMesReceived ( CVector<CChannelInfo> vecChanInfo )
+void CClient::OnMuteStateHasChangedReceived ( int iServerChanID, bool bIsMuted )
+{
+    // map iChanID from server channel ID to client channel ID
+    int iChanID = FindClientChannel ( iServerChanID, false );
+
+    emit MuteStateHasChangedReceived ( iChanID, bIsMuted );
```

Comment:
This causes
```cpp
 vecpChanFader[static_cast<size_t> ( iChannelIdx )]->SetRemoteFaderIsMute ( bIsMute );
```
so if `iChanID` is `INVALID_INDEX`, the client will crash.

Reply:
Good catch! Highly unlikely in practice, but we shouldn't allow it. I'll protect the `emit` with a condition.